speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -161,7 +166,70 @@ static void SetUserId(void*)
     GfuiEditboxSetString(s_scrHandle, m_userIdControl, m_userId);
 }
 
+/// @brief Saves the settings into the ResearcherMenu.xml file
+static void SaveSettingsToDisk() {
+    // Open file parameter
+    std::string strPath("data/menu/ResearcherMenu.xml");
+    char buf[512];
+    sprintf(buf, "%s%s", GfDataDir(), strPath.c_str());
+    void* readParam = GfParmReadFile(buf, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Save task settings to xml file
+    GfParmSetStr(readParam, "dynamic controls/CheckboxTaskLaneKeeping", "checked", "no");
+    GfParmSetStr(readParam, "dynamic controls/CheckboxTaskSpeedControl", "checked", "no");
+    switch (m_task)
+    {
+        case 1:
+            GfParmSetStr(readParam, "dynamic controls/CheckboxTaskLaneKeeping", "checked", "yes");
+            break;
+        case 2:
+            GfParmSetStr(readParam, "dynamic controls/CheckboxTaskSpeedControl", "checked", "yes");
+            break;
+    }
+
+    // Save indicator settings to xml file
+    const char* audioSetting = m_indicators.Auditory ? "yes" : "no";
+    GfParmSetStr(readParam, "dynamic controls/CheckboxIndicatorAuditory", "checked", audioSetting);
+    const char* visualSetting = m_indicators.Visual ? "yes" : "no";
+
+    // Save intervention type settings to xml file
+    GfParmSetStr(readParam, "dynamic controls/CheckboxIndicatorVisual", "checked", visualSetting);
+    GfParmSetStr(readParam, "dynamic controls/CheckboxTypeNoSignals", "checked", "no");
```

Comment:
Misplaced a GfParmSetStr
```suggestion
    // Save indicator settings to xml file
    const char* audioSetting = m_indicators.Auditory ? "yes" : "no";
    GfParmSetStr(readParam, "dynamic controls/CheckboxIndicatorAuditory", "checked", audioSetting);
    const char* visualSetting = m_indicators.Visual ? "yes" : "no";
    GfParmSetStr(readParam, "dynamic controls/CheckboxIndicatorVisual", "checked", visualSetting);

    // Save intervention type settings to xml file
    GfParmSetStr(readParam, "dynamic controls/CheckboxTypeNoSignals", "checked", "no");
```